mountpoint-s3/src/prefix.rs

```diff
@@ -0,0 +1,65 @@
+use std::fmt::Display;
+use std::str::FromStr;
+
+use anyhow::anyhow;
```

Comment:
I'd build a real error type here instead of using anyhow:
```rust
use thiserror::Error;

#[derive(Error, Debug)]
pub enum PrefixError {
    #[error("prefixes must end in '/'")]
    MissingFinalDelimiter,
}
```
and then
```rust
impl Prefix {
    pub fn new(prefix: &str) -> Result<Self, PrefixError> {
        ....
```

(Roughly speaking, our approach has been "real error types everywhere except `main.rs`", which means anyone using this crate as a library gets real error types)